pkg/wallet/wallet.go

```diff
@@ -986,14 +988,29 @@ func (c *Wallet) validateProofOption(authToken string, opts *ProofOptions, metho
 func (c *Wallet) validateVerificationMethod(didDoc *did.Doc, opts *ProofOptions,
 	relationship did.VerificationRelationship) error {
 	vms := didDoc.VerificationMethods(relationship)[relationship]
+	kt := kms.ED25519Type
 
 	for _, vm := range vms {
 		if opts.VerificationMethod == "" {
 			opts.VerificationMethod = vm.VerificationMethod.ID
-			return nil
 		}
 
 		if opts.VerificationMethod == vm.VerificationMethod.ID {
+			switch vm.VerificationMethod.Type {
+			case verifiablec.Ed25519VerificationKey2018:
+			case verifiablec.Bls12381G2Key2020:
+				kt = kms.BLS12381G2Type
+			case verifiablec.JSONWebKey2020:
+				kt = verifiablec.KmsKeyTypeByJWKCurve(vm.VerificationMethod.JSONWebKey().Crv)
+			}
+
+			kid, err := localkms.CreateKID(vm.VerificationMethod.Value, kt)
```

Comment:
KID shouldn't be computed here in wallet and shouldn't replace the option KID. If In case client wants to use custom KID then it should be computed outside and passed as an option.

Main reason for this is that, different DID method implementations use their own way of preparing key IDs. 
Creating and using  our own key IDs will break the interop in this wallet. I guess yet there is no standard for creating key IDs in DIDs .

Imagine a case where someone configured VCWallet with their own VDR and remote KMS (webkms), then this change will force remote kms implementation to use this `localkms.CreateKID(...)` logic to create their key IDs.

You can refer [this test in trustbloc](https://github.com/trustbloc/agent-sdk/blob/main/cmd/wallet-js-sdk/test/specs/credential/credential-manager.spec.js#L114-L145) where `wallet.CreateKeyPair()` is used to create keys for a DID to issue/verify credentials.

We are yet to add bddtests for vcwallet in aries. I would love to have chat if you want to discuss more about this.